### DISCUSSION TITLE
Qt package: commercial and -no-opengl

### DISCUSSION THREAD

**Discussion by srebrnyp** [microsoft/vcpkg#30255] [Q&A] [answered]
Hi,

Two general questions:
I want to configure Qt package to use commercial version, its possible? how to do that?
And also how to configure Qt to not use opengl, just to pass -no-opengl switch?

Best
Paweł

**Comment by Osyotr** [comment, marked as answer]
Assuming you want to build qt5, the easiest solution is to create an overlay port and edit these lines:
https://github.com/microsoft/vcpkg/blob/master/ports/qt5-base/portfile.cmake#L85 - replace `-opensource` with `-commercial`
https://github.com/microsoft/vcpkg/blob/master/ports/qt5-base/portfile.cmake#L247-L251 - replace `-opengl dynamic` with `-no-opengl`

**Reply by srebrnyp** [reply]
Thanks for answer.
And additional question, is there a way to install qt version that are not presented in: https://download.qt.io/official_releases
For example there is no version 5.15.10, this version is only available after login to qt.io commercial side.
In basic, what would be the best way to adopt scripts to work with full version of downloaded sources (in one package ex: qt-everywhere-src-5.15.10.tar.xz).

**Reply by Osyotr** [reply]
Here's an instruction I made for people who want to update qt5: https://github.com/microsoft/vcpkg/pull/24660#issuecomment-1124329485
If you have access to submodules, just replace URL [here](https://github.com/microsoft/vcpkg/blob/master/ports/qt5-base/cmake/qt_download_submodule.cmake#L17-L20) and follow the instruction.
For single archive I think you can also replace this URL but skip hashes stuff from the instruction. Keep in mind that vcpkg will unpack this archive over and over for every qt5-\* port. This is a massive waste.
